### ISSUE TITLE
Missing JNI Exception Checks

### ISSUE THREAD

**Issue by chris-smith-zocdoc** [microsoft/LightGBM#2360]
I'm attempting to track down a memory corruption bug that is occurring during training of LightGBM on spark (using https://github.com/Azure/mmlspark) 

But when I enable the jvm options `-Xcheck:jni -verbose:jni` I am seeing an excessive number of warnings suggesting that errors aren't being checked in the jni code. 

I haven't written jni before, but only call to `CallObjectMethod` is [here](https://github.com/microsoft/LightGBM/blob/master/swig/lightgbmlib.i#L157-L158)

A quick google search indicates that we should be using `env->ExceptionCheck()` or `env-> ExceptionOccurred()` after calling methods that can throw exceptions https://www.developer.com/java/data/exception-handling-in-jni.html

## Environment info

Databricks 5.5 ML Runtime

Operating System:
```
4.4.0-1087-aws

Ubuntu 16.04.6 LTS
```

CPU/GPU model:
```
AWS r4.xlarge

Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
```

LightGBM version or commit hash:
```
com.microsoft.ml.lightgbm lightgbmlib-2.2.350.jar
```

JVM Info
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-8u212-b03-0ubuntu1.16.04.1-b03)
OpenJDK 64-Bit Server VM (build 25.212-b03, mixed mode)
```

## Error message

```
WARNING in native method: JNI call made without checking exceptions when required to from CallObjectMethodV
	at com.microsoft.ml.lightgbm.lightgbmlibJNI.LGBM_DatasetCreateFromCSRSpark(Native Method)
	at com.microsoft.ml.lightgbm.lightgbmlib.LGBM_DatasetCreateFromCSRSpark(lightgbmlib.java:286)
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:246)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:79)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:258)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:411)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:85)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:85)
	at org.apache.spark.rdd.RDDBarrier$$anonfun$mapPartitions$1$$anonfun$apply$1.apply(RDDBarrier.scala:51)
	at org.apache.spark.rdd.RDDBarrier$$anonfun$mapPartitions$1$$anonfun$apply$1.apply(RDDBarrier.scala:51)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:60)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:340)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:304)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:139)
	at org.apache.spark.scheduler.Task.run(Task.scala:112)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$13.apply(Executor.scala:497)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1526)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:503)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

## Steps to reproduce

1. Enable jvm options `spark.executor.extraJavaOptions -Xcheck:jni -verbose:jni`
2. Begin training a model on sparse data

**Comment by StrikerRUS** [comment]
ping @imatiach-msft
Feel free to transfer this issue to https://github.com/Azure/mmlspark repo if you feel it should be there.

**Comment by StrikerRUS** [comment]
Fixed via #2405.
